OWML.Launcher/SocketListener.cs

```diff
@@ -73,23 +78,40 @@ private void ListenToSocket()
                     ProcessMessage(bytes, i);   
                 }
 
+                ConsoleUtils.WriteByType(MessageType.Success, "Closing client!");
                 client.Close();
             }
         }
 
         private void ProcessMessage(byte[] bytes, int count)
         {
-            var json = Encoding.UTF8.GetString(bytes, 0, count);
-
-            var data = JsonConvert.DeserializeObject<SocketMessage>(json);
-
-            if (data.Type == MessageType.Quit)
+            var message = Encoding.UTF8.GetString(bytes, 0, count);
+            var jsons = message.Split('\n');
```

Comment:
What if a message inside a json contains a \n

Reply:
it shouldn't

Reply:
inside messages themselves they are written as "\\n" which != '\n', and jsons are one-liners

Reply:
so, unless you add newlines into json yourself it will work fine 😝

Reply:
and I considered secret combination to be a bit iffy

Reply:
> inside messages themselves they are written as "\\n"

Always? Whether log came from OWML or Unity?

Reply:
yes, it does that when converting to json string (and yes I tested it)

Reply:
If you say so ;)

Reply:
as it works with both OWML message (one from Owo about changed settings) and Unity message (when testing with stack overflow trace)